### PR TITLE
Add support for Kubernetes stacks

### PR DESCRIPTION
**Pull request by adamconnelly**
## Description of the change

- Added support for Kubernetes stacks.
- Simplified the stack resource tests for CloudFormation and Pulumi to avoid checking every attribute. That's covered by other tests, and it makes them less clear.
- Added data source tests for CloudFormation, Kubernetes and Pulumi and fixed a bug in the CloudFormation settings for the stack data source where it was setting `entry_template_name` instead of `entry_template_file`.
- Refactored the stack examples to avoid confusion between stacks creating Kubernetes resources vs stacks using the Kubernetes provider.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (non-breaking change that adds documentation)

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review

- [x] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] Pull Request is no longer marked as "draft"
- [ ] Reviewers have been assigned
- [ ] Changes have been reviewed by at least one other engineer
- [x] The target branch is `future` unless the change is going directly into production

### CODE REVIEW

**Comment by marcinwyszynski** [comment]
Task linked: [CU-29axerq Add support for Kubernetes Stacks to the terraform provider.](https://app.clickup.com/t/29axerq)

File: spacelift/data_stack.go [hunk 1]
```diff
@@ -220,6 +220,20 @@ func dataStack() *schema.Resource {
...
 			},
+			"kubernetes": {
+				Type:        schema.TypeList,
+				Description: "Kubernetes-specific configuration. Presence means this Stack is a Kubernetes Stack.",
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"namespace": {
+							Type:        schema.TypeString,
+							Description: "The Kubernetes namespace to deploy resources to. When not specified, resources with no explicit namespace specified will be deployed to the default namespace.",
```
Comment (cube2222) [inline spacelift/data_stack.go:231]: This should read a bit different. The namespace should be left unspecified for multi-namespace stacks. 

The frontend copy is:
```
Namespace of Kubernetes cluster to run commands on. Leave empty for multi-namespace Stacks.
```

File: spacelift/resource_stack.go [hunk 2]
```diff
@@ -264,6 +264,22 @@ func resourceStack() *schema.Resource {
...
+				Type:          schema.TypeList,
+				ConflictsWith: []string{"cloudformation", "pulumi", "terraform_version", "terraform_workspace"},
+				Description:   "Kubernetes-specific configuration. Presence means this Stack is a Kubernetes Stack.",
+				Optional:      true,
+				MaxItems:      1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"namespace": {
+							Type:        schema.TypeString,
+							Description: "The Kubernetes namespace to deploy resources to. When not specified, resources with no explicit namespace specified will be deployed to the default namespace.",
```
Comment (cube2222) [inline spacelift/resource_stack.go:277]: As above.

**Review by cube2222 — APPROVED** [review]
with nit

File: spacelift/data_stack.go (hunk 1 above)
Comment (adamconnelly) [reply spacelift/data_stack.go]: Ok - I've updated it to use the same copy.

File: spacelift/resource_stack.go (hunk 2 above)
Comment (adamconnelly) [reply spacelift/resource_stack.go]: Same.

**Pull request merged**